src/ast/codegen_llvm.cpp

```diff
@@ -1004,6 +1004,24 @@ void CodegenLLVM::visit(Call &call)
     auto arg = call.vargs->at(0);
     auto scoped_del = accept(arg);
   }
+  else if (call.func == "macaddr")
+  {
+    // MAC addresses are presented as char[6]
+    AllocaInst *buf = b_.CreateAllocaBPFInit(call.type, "macaddr");
+    b_.CREATE_MEMSET(buf, b_.getInt8(0), 6, 1);
+    auto macaddr = call.vargs->front();
+    auto scoped_del = accept(macaddr);
+
+    b_.CreateProbeRead(ctx_,
```

Comment:
IIRC arrays can end up on stack already so this would probe read from the stack in that case

Reply:
How should I check for that case? I see that `shouldBeOnStackAlready` doesn't include arrays, so it wouldn't be enough to branch on that. But I'm thinking that we need to branch on something and then store (instead of probe read) if the arg is already on the stack.

Reply:
Yes it would have to load/memcpy instead of probe reading.

I dont know the best way to check, arrays need some love. I'd be ok with this probe reading for now too, can always fix it in the future.